ingest_graph_validator/hydrators/ingest_hydrator.py

```diff
@@ -109,31 +121,32 @@ def get_edges(self):
             'derivedFiles': "files",
         }
 
-        for entity_uuid, entity in self._entities.items():
+        entity_num = 0
+        for entity in self._entities:
             for relationship_type in relationship_map.keys():
                 if relationship_type in entity['links']:
-                    relationships = self._ingest_api.get_all(
-                        entity['links'][relationship_type]['href'],
-                        relationship_map[relationship_type]
-                    )
-
+                    url = entity['links'][relationship_type]['href']
+                    entity_type = relationship_map[relationship_type]
+                    relationships = self._ingest_api.get_all(url, entity_type)
```

Comment:
We might also want to review the page_size in get_all, it's currently 100. From my testing of "export to spreadsheet" (I'm not sure yet, I did some changes locally and couldn't isolate which exactly improved the API calls) but I think reducing the data being the fetch from ingest-core prevents it from crashing (I don't know why 100 is too much, we should do some more testing to figure out why) but yeah, I tried setting it to 10 and from my testing of submission `1e806c65-d578-40d1-868b-c8965ac29b8b` in dev, it doesn't seem to make core crash so far. The only thing I'm monitoring now is the memory consumption of the ingest-client and I also changed some parts to use the generators.

Reply:
Please note those testing are all for "export to spreadsheet" but may also be applicable in this case.